src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,278 @@
+
+
+    def pics_TC_BRBINFO_4_1(self) -> list[str]:
+        """ This function returns a list of PICS for this test case that must be True for the test to be run"""
+        pics = [
+            "BRBINFO.C.80",
+        ]
+        return pics
+
+    def _ask_for_vendor_commissioniong_ux_operation(self, discriminator, setupPinCode):
+        self.wait_for_user_input(
+            prompt_msg=f"Using the DUT vendor's provided interface, commission the ICD device using the following parameters:\n"
+                        f"- discriminator: {discriminator}\n"
+                        f"- setupPinCode: {setupPinCode}\n"
+                        f"If using FabricSync Admin, you may type:\n"
+                        f">>> pairing onnetwork 111 {setupPinCode}")
+
+    def _display_message_and_wait_for_user_input(self, message):
+        self.wait_for_user_input(prompt_msg=message)
+
+    async def _send_keep_active_command(self, duration) -> int:
+        logging.info("Mock sending keep active command")
+        keep_active = await self.default_controller.SendCommand(nodeid=self.dut_node_id, endpoint=kICDBridgedEndpointId, payload=Clusters.Objects.BridgedDeviceBasicInformation.Commands.KeepActive(stayActiveDuration=duration))
+        return keep_active
+
+    async def _wait_for_active_changed_event(self, timeout) -> int:
+        try:
+            promised_active_duration = self.q.get(block=True, timeout=timeout)
+            logging.info(f"PromisedActiveDuration: {promised_active_duration}")
+            return promised_active_duration
+        except queue.Empty:
+            asserts.fail("Timeout on event")
+
+
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+        app = os.path.join(pathlib.Path(__file__).resolve().parent, '..','..','out', 'linux-x64-lit-icd', 'lit-icd-app')
+
+        self.kvs = f'kvs_{str(uuid.uuid4())}'
+        self.port = 5543
+        discriminator = 3850
+        passcode = 20202021
+        app_args = f'--secured-device-port {self.port} --discriminator {discriminator} --passcode {passcode} --KVS {self.kvs} ' + \
+          '--secured-device-port 5545'
+        cmd = f'{app} {app_args}'
+
+        logging.info("Starting ICD Server App")
+        self.app_process = subprocess.Popen(cmd, bufsize=0, shell=True)
+        logging.info("ICD started")
+        time.sleep(3)
+
+        logging.info("Commissioning of ICD to fabric one (TH)")
+        self.icd_nodeid = 1111
+
+        await self.default_controller.CommissionOnNetwork(nodeId=self.icd_nodeid, setupPinCode=passcode, filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=discriminator)
+
+        logging.info("Commissioning of ICD to fabric two (DUT)")
+        params = await self.openCommissioningWindow(dev_ctrl=self.default_controller, node_id=self.icd_nodeid)
+
+        self._ask_for_vendor_commissioniong_ux_operation(params.randomDiscriminator, params.commissioningParameters.setupPinCode)
```

Comment:
so this is the vendor commissioning the app that the TH just pulled up? Do all the DUTs being tested here have to support ICD? Does it need to be a specific device type?

Reply:
Yes, I open the commissioning window for the ICD server and ask the DUT vendor to commission it, giving the command for convenience if the tester's DUT is using something close to the sample FabricSync bridge interface.

> Do all the DUTs being tested here have to support ICD
Yes

> Does it need to be a specific device type?
You mean the ICD?